hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/net/AsyncServerSocketMetrics.java

```diff
@@ -16,47 +16,41 @@
 
 package com.hazelcast.internal.tpcengine.net;
 
-import com.hazelcast.internal.tpcengine.util.UnsafeLocator;
-import sun.misc.Unsafe;
-
-import java.lang.reflect.Field;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
 
 /**
  * Contains the metrics for an {@link AsyncServerSocket}.
  */
 @SuppressWarnings("checkstyle:ConstantName")
 public class AsyncServerSocketMetrics {
 
-    private static final Unsafe UNSAFE = UnsafeLocator.UNSAFE;
-    private static final long OFFSET_accepted;
-    private volatile long accepted;
+    private static final VarHandle ACCEPTED;
 
     static {
         try {
-            OFFSET_accepted = getOffset("accepted");
-        } catch (NoSuchFieldException e) {
-            throw new RuntimeException(e);
+            MethodHandles.Lookup l = MethodHandles.lookup();
+            ACCEPTED = l.findVarHandle(AsyncServerSocketMetrics.class, "accepted", long.class);
+        } catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError(e);
         }
     }
 
-    private static long getOffset(String fieldName) throws NoSuchFieldException {
-        Field field = AsyncServerSocketMetrics.class.getDeclaredField(fieldName);
-        return UNSAFE.objectFieldOffset(field);
-    }
+    private volatile long accepted;
 
     /**
      * Returns the number of accepted sockets.
      *
      * @return the number of accepted sockets.
      */
     public long accepted() {
-        return accepted;
+        return (long) ACCEPTED.getOpaque(this);
     }
 
     /**
      * Increases the number of accepted sockets by 1.
      */
     public void incAccepted() {
-        UNSAFE.putOrderedLong(this, OFFSET_accepted, accepted + 1);
+         ACCEPTED.setOpaque(this, (long) ACCEPTED.getOpaque(this) + 1);
```

Comment:
Why not just use atomic long for this? Is using opaque even wort here?

Reply:
Opaque is better for high-performance progress indicators because you will not get CPU memory fences and will restrict the JIT less.

Reply:
Valid point. I made the code consistent with the AsyncSocketMetrics and these methods can be called very frequently.

Reply:
Yeah no worries, I also guessed like that, hence I approved. Just wanted to be sure I didn't missed anything.